portal/models/qb_timeline.py

```diff
@@ -384,7 +384,9 @@ def next_qbd(self):
                 trace("Finished cur RP with remaining QBs in next")
                 self.cur_start = self.nxt_start
                 self.transition()
-            elif self.cur_start != self.nxt_start:
+            elif self.cur_start > self.nxt_start + relativedelta(months=1):
```

Comment:
A bit worried about patients falling between RPs, but I'm sure you've thought of it :)

Reply:
I'm more that a bit worried, this overly complex set of requirements is difficult to describe in code.  We're shifting designs soon.  Hopefully it will be easier to maintain.